### PR TITLE
handles click propagate for route decorator in kn service

### PR DESCRIPTION
**Pull request by invincibleJai**
- handles click propagate for route decorator in kn service

Tracks: https://jira.coreos.com/browse/ODC-2218

Gif:

ezgif com-video-to-gif (8)

### CODE REVIEW

**Comment by christianvogt** [comment]
I think this is better to be handled generically at the decorator level. There shouldn't be any need to propagate a click through the decorator if there is an href. Probably even if there isn't an href as well.

File: frontend/packages/dev-console/src/components/topology/shapes/Decorator.tsx [hunk 1]
```diff
@@ -27,7 +27,13 @@ const Decorator: React.FunctionComponent<DecoratorTypes> = ({
   circleRef,
 }) => {
   const decorator = (
-    <g className="odc-decorator" onClick={onClick}>
+    <g
+      className="odc-decorator"
+      onClick={(e) => {
+        e.stopPropagation();
+        onClick ? onClick() : null;
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/topology/shapes/Decorator.tsx]: nit: 
```suggestion
        onClick && onClick();
```

**Pull request merged**